e2e-tests/spring-rest-openapi-v2/src/main/java/com/foo/rest/examples/spring/authenticatedswaggeraccess/AuthenticatedSwaggerApplicationWebSecurityConfig.java

```diff
@@ -0,0 +1,61 @@
+package com.foo.rest.examples.spring.authenticatedswaggeraccess;
+
+import org.evomaster.client.java.controller.AuthUtils;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+@Configuration
+@EnableWebSecurity
+public class AuthenticatedSwaggerApplicationWebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    // security configuration
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                //  .antMatchers("/v2/api-docs").permitAll()
+                .antMatchers("/v2/api-docs").hasRole("AUTH_USER")
+                .antMatchers(HttpMethod.GET, "/endpoint1").permitAll()
+                .antMatchers(HttpMethod.PUT, "/endpoint2").hasAnyRole("AUTH_USER")
```

Comment:
wasn't `endpoint2` a GET and not a PUT?

Reply:
Hi Andrea,

Thanks for your comment. I have fixed this one.